go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -423,69 +371,137 @@ func (oc *Controller) createEgressFirewallRules(priority int, match, action, ext
+	//		externalID, err)
+	//}
+
+	// add it's UUID to the necessary logical switches
+	opModels := []libovsdbops.OperationModel{}
+	switches := []*nbdb.LogicalSwitch{}
+	for _, logicalSwitchName := range logicalSwitches {
+		lsn := logicalSwitchName
+		lsw := nbdb.LogicalSwitch{Name: lsn}
+		switches = append(switches, &lsw)
+		opModels = append(opModels, []libovsdbops.OperationModel{
+			{
+				Model:          &lsw,
+				ModelPredicate: func(ls *nbdb.LogicalSwitch) bool { return ls.Name == lsn },
+				OnModelMutations: []interface{}{
+					&lsw.ACLs,
+				},
+				ErrNotFound: true,
+				BulkOp:      true,
+			},
+		}...)
+	}
 
-		} else {
-			for _, uuid := range strings.Fields(uuids) {
-				_, stderr, err := txn.AddOrCommit([]string{"add", "logical_switch", logicalSwitch, "acls", uuid})
-				if err != nil {
-					return fmt.Errorf("failed to commit db changes for egressFirewall stderr: %q, err: %+v", stderr, err)
+	foundACLs := []nbdb.ACL{}
+	opModels = append([]libovsdbops.OperationModel{
+		{
+			Model:          egressFirewallACL,
+			ModelPredicate: func(acl *nbdb.ACL) bool { return libovsdbops.IsEquivalentACL(acl, egressFirewallACL) },
+			ExistingResult: &foundACLs,
+			DoAfter: func() {
+				var uuid string
+				if len(foundACLs) == 1 {
+					uuid = foundACLs[0].UUID
+				} else {
+					uuid = egressFirewallACL.UUID
 				}
-			}
-		}
+				for _, lsw := range switches {
+					lsw.ACLs = []string{uuid}
+				}
+			},
+		},
+	}, opModels...)
+
+	if _, err := oc.modelClient.CreateOrUpdate(opModels...); err != nil {
+		return fmt.Errorf("failed to create egressFirewall ACL in ns %s and add to logical switches %v err: %v",
+			externalID, logicalSwitches, err)
 	}
+
 	return nil
 }
 
 // deleteEgressFirewallRules delete the specific logical router policy/join switch Acls
-func (oc *Controller) deleteEgressFirewallRules(externalID string, txn *util.NBTxn) error {
-	logicalSwitches := []string{}
-	if config.Gateway.Mode == config.GatewayModeLocal {
-		nodes, err := oc.watchFactory.GetNodes()
-		if err != nil {
-			return fmt.Errorf("unable to setup egress firewall ACLs on cluster nodes, err: %v", err)
-		}
-		for _, node := range nodes {
-			logicalSwitches = append(logicalSwitches, node.Name)
+func (oc *Controller) deleteEgressFirewallRules(externalID string) error {
+	// Find ACLs for a given egressFirewall
+	egressFirewallACLLookupFcn := func(item *nbdb.ACL) bool {
+		firewallMatch := false
+		if ns, ok := item.ExternalIDs["egressFirewall"]; ok {
+			if ns == externalID {
+				firewallMatch = true
+			}
 		}
-	} else {
-		logicalSwitches = []string{types.OVNJoinSwitch}
+		return firewallMatch
 	}
-	sort.Strings(logicalSwitches)
-	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading", "--columns=_uuid", "--format=table", "find", "ACL",
-		fmt.Sprintf("external-ids:egressFirewall=%s", externalID))
+
+	egressFirewallACLs, err := libovsdbops.FindACLs(oc.modelClient.GetClient(), egressFirewallACLLookupFcn)
```

Comment:
Would prefer `FindEgressFirewallACLsOnNamespace` or otherwise `FindACLsByPredicate`. It's kinda having an inventory of search criteria of things rather than using predicates by value all around and not reuse, I would only use predicates for very specific one time only weird kind of thing.

Reply:
Done, added a helper function for the case